### PR TITLE
Feat.lint obsolete vars

### PR DESCRIPTION
**Pull request by wxtim**
Another item from Oliver's wishlist at https://github.com/cylc/cylc-flow/issues/4917:

Search for deprecated items from [this list](https://cylc.github.io/cylc-doc/stable/html/user-guide/writing-workflows/runtime.html#task-event-template-variables)

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/scripts/lint.py [hunk 1]
```diff
@@ -106,6 +106,29 @@
...
 
 
+DEPRECATED_STRING_TEMPLATES = {
+    'suite': 'workflow',
+    'suite_uuid': 'uuid',
+    'batch_sys_name': 'job_runner_name',
+    'batch_sys_job_id': 'job_id',
+    'user@host': 'platform_name',
+    'task_url': 'URL from task metadata',
+    'workflow_url': 'workflow_URL from workflow metadata',
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/scripts/lint.py:116]: You can set these items in 

```
# flow.cylc
[meta]
    workflow_url = "www.hat.ever"

[runtime]
    [[task]]
        [[[meta]]]
            task_url = "some://url"

```
And they will be available.

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/scripts/lint.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/scripts/lint.py:116]: I agree. I'll have a think about how this might work.

[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/scripts/lint.py]: @MetRonnie  - Have another look, I should think that it's better now.

**Review by MetRonnie — APPROVED** [review]

**Review by MetRonnie — CHANGES_REQUESTED** [review]
Ah wait, got test failures

**Comment by wxtim** [comment]
> Ah wait, got test failures

Tests earning their keep. Hopefully Will pass now.

**Review by MetRonnie — APPROVED** [review]
:+1:

**Review by markgrahamdawson — APPROVED** [review]

**Pull request merged**